dc/s2n-quic-dc/src/task/waker/set.rs

```diff
@@ -0,0 +1,92 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+use super::worker;
+use std::{
+    sync::{Arc, Mutex},
+    task::{Wake, Waker},
+};
+
+mod bitset;
+use bitset::BitSet;
+
+#[derive(Default)]
+pub struct Set {
+    state: Arc<State>,
+    ready: BitSet,
+}
+
+impl Set {
+    /// Updates the root waker
+    pub fn update_root(&self, waker: &Waker) {
+        self.state.root.update(waker);
+    }
+
+    /// Registers a waker with the given ID
+    pub fn waker(&mut self, id: usize) -> Waker {
+        // reserve space in the locally ready set
+        self.ready.resize_for_id(id);
```

Comment:
It is fixed after we create all of the Wakers. This just made it so you don't have to specify the capacity up front. In the critical path, this isn't used at all, since those wakers are cached in the slot.